# See Also: https://github.com/akiradeveloper/dm-writeboost/issues/165
diff -aurN dm-writeboost-2.2.6.org/src/dm-writeboost.h dm-writeboost-2.2.6/src/dm-writeboost.h
--- dm-writeboost-2.2.6.org/src/dm-writeboost.h	2016-09-19 06:15:04.000000000 +0000
+++ dm-writeboost-2.2.6/src/dm-writeboost.h	2017-01-06 13:40:46.943889413 +0000
@@ -37,6 +37,12 @@
 #include <linux/dm-io.h>
 #include <linux/dm-kcopyd.h>
 
+/* we use RHEL_RELEASE_VERSION to compile with RHEL/CentOS 7.3's kernel  */
+#ifndef RHEL_RELEASE_CODE
+#define RHEL_RELEASE_CODE 0
+#define RHEL_RELEASE_VERSION(a,b) (((a) << 8) + (b))
+#endif
+
 /*----------------------------------------------------------------------------*/
 
 #define SUB_ID(x, y) ((x) > (y) ? (x) - (y) : 0)
diff -aurN dm-writeboost-2.2.6.org/src/dm-writeboost-target.c dm-writeboost-2.2.6/src/dm-writeboost-target.c
--- dm-writeboost-2.2.6.org/src/dm-writeboost-target.c	2016-09-19 06:15:04.000000000 +0000
+++ dm-writeboost-2.2.6/src/dm-writeboost-target.c	2017-01-06 13:40:14.581859591 +0000
@@ -674,7 +674,7 @@
 	PBD_READ_SEG = 2,
 };
 
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(4,6,0)
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(4,6,0) || RHEL_RELEASE_CODE >= RHEL_RELEASE_VERSION(7,3))
 #define PER_BIO_DATA_SIZE per_io_data_size
 #else
 #define PER_BIO_DATA_SIZE per_bio_data_size
