diff -aurN writeboost-1.20160718-37dc927f969c34947f2de49afd5bb4cc95078768.org/writeboost.service writeboost-1.20160718-37dc927f969c34947f2de49afd5bb4cc95078768/writeboost.service
--- writeboost-1.20160718-37dc927f969c34947f2de49afd5bb4cc95078768.org/writeboost.service	2017-01-10 20:44:10.652538605 +0900
+++ writeboost-1.20160718-37dc927f969c34947f2de49afd5bb4cc95078768/writeboost.service	2017-01-10 20:44:36.239537662 +0900
@@ -7,7 +7,7 @@
 
 [Service]
 Type=oneshot
-RemainAfterExit=no
+RemainAfterExit=yes
 TimeoutStartSec=600
 ExecStart=/sbin/writeboost
 ExecStop=/sbin/writeboost -u
